### PR TITLE
tgui-port: Ports round (pressure) gauge

### PR DESCRIPTION
**Pull request by ZeWaka**
[feat]

## About the PR 

Ports https://github.com/tgstation/tgstation/pull/55230

<details>
  <summary>More Images</summary>

image

</details>

also cleaned up the data sending like the other PR did

## Why's this needed? 

pretty

also fixes paper embedding onto canister bombs

### CODE REVIEW

**Comment by ThePotato97** [comment]
Can you make it this width?
image

File: tgui/packages/tgui/components/RoundGauge.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+          ...rest,
+        })}>
+        <svg
+          viewBox="0 0 100 50">
+          {alertAfter && (
+            <g className={classes([
+              'RoundGauge__alert',
+              alertColor ? `active RoundGauge__alert--${alertColor}` : '',
+            ])}>
+              <path d="M48.211,14.578C48.55,13.9 49.242,13.472 50,13.472C50.758,13.472 51.45,13.9 51.789,14.578C54.793,20.587 60.795,32.589 63.553,38.106C63.863,38.726 63.83,39.462 63.465,40.051C63.101,40.641 62.457,41 61.764,41C55.996,41 44.004,41 38.236,41C37.543,41 36.899,40.641 36.535,40.051C36.17,39.462 36.137,38.726 36.447,38.106C39.205,32.589 45.207,20.587 48.211,14.578ZM50,34.417C51.426,34.417 52.583,35.574 52.583,37C52.583,38.426 51.426,39.583 50,39.583C48.574,39.583 47.417,38.426 47.417,37C47.417,35.574 48.574,34.417 50,34.417ZM50,32.75C50,32.75 53,31.805 53,22.25C53,20.594 51.656,19.25 50,19.25C48.344,19.25 47,20.594 47,22.25C47,31.805 50,32.75 50,32.75Z" />
```
Comment (ThePotato97) [inline tgui/packages/tgui/components/RoundGauge.js:75]: Why not use exclamation-triangle icon from fontawesome?

Comment (ZeWaka) [reply tgui/packages/tgui/components/RoundGauge.js:75]: Original PR, but I'll look into it

Comment (ZeWaka) [reply tgui/packages/tgui/components/RoundGauge.js:75]: Has rendering problems, probably due to being inside a svg. It disappears when opening it after you have another window with a gauge open. Weird.

**Comment by ZeWaka** [comment]
> Can you make it this width?

Done.

File: tgui/packages/tgui/interfaces/common/PortableAtmos.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import { Fragment } from 'inferno';
```
Comment (ThePotato97) [inline tgui/packages/tgui/interfaces/common/PortableAtmos.js:1]:
```suggestion
```
unused import

Comment (ZeWaka) [reply tgui/packages/tgui/interfaces/common/PortableAtmos.js:1]: nope, it's used.

File: code/modules/atmospherics/portable/canister.dm [hunk 3]
```diff
@@ -387,49 +387,52 @@
...
+				"time" = src.det.part_fs.time * 10, // using tenths of a second on the client
+				"trigger" = src.det.trigger ? src.det.trigger.name : null,
+			)
+		)
 
 /obj/machinery/portable_atmospherics/canister/ui_static_data(mob/user)
 	var/list/static_data = list()
 
+	static_data["minRelease"] = PORTABLE_ATMOS_MIN_RELEASE_PRESSURE
+	static_data["maxRelease"] = PORTABLE_ATMOS_MAX_RELEASE_PRESSURE
```
Comment (mordent-goonstation) [inline code/modules/atmospherics/portable/canister.dm:427]:
```suggestion
	. = list(
		"minRelease" = PORTABLE_ATMOS_MIN_RELEASE_PRESSURE,
		"maxRelease" = PORTABLE_ATMOS_MAX_RELEASE_PRESSURE
	)
```
And other associated changes to work with it? Not requiring this change.

File: tgui/packages/tgui/styles/components/RoundGauge.scss [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+ $fg-map: colors.$fg-map !default;
+ $ring-color: #6a96c9 !default;
+
+ .RoundGauge {
+   font-size: 1rem;
+   width: 2.6em;
+   height: 1.3em;
+   margin: 0 auto;
+   margin-bottom: 0.2em;
```
Comment (mordent-goonstation) [inline tgui/packages/tgui/styles/components/RoundGauge.scss:18]:
```suggestion
   margin: 0 auto 0.2em;
```

File: tgui/packages/tgui/components/RoundGauge.js [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+    style,
+    ...rest
+  } = props;
+
+  const scaledValue = scale(
+    value,
+    minValue,
+    maxValue);
+  const clampedValue = clamp01(scaledValue);
+  let scaledRanges = ranges ? {} : { "primary": [0, 1] };
```
Comment (mordent-goonstation) [inline tgui/packages/tgui/components/RoundGauge.js:38]:
```suggestion
  const scaledRanges = ranges ? {} : { "primary": [0, 1] };
```

File: tgui/packages/tgui/components/RoundGauge.js [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+  if (ranges)
+  { Object.keys(ranges).forEach(x => {
+    const range = ranges[x];
+    scaledRanges[x] = [
+      scale(range[0], minValue, maxValue),
+      scale(range[1], minValue, maxValue),
+    ];
+  }); }
```
Comment (mordent-goonstation) [inline tgui/packages/tgui/components/RoundGauge.js:46]:
```suggestion
  if (ranges) {
    Object.keys(ranges).forEach(x => {
      const range = ranges[x];
      scaledRanges[x] = [
        scale(range[0], minValue, maxValue),
        scale(range[1], minValue, maxValue),
      ];
    });
  }
```

**Review by mordent-goonstation — CHANGES_REQUESTED** [review]

File: code/modules/atmospherics/portable/canister.dm (hunk 3 above)
Comment (ZeWaka) [reply code/modules/atmospherics/portable/canister.dm:427]: this change doesn't require any associated changes though?

File: code/modules/atmospherics/portable/canister.dm [hunk 7]
```diff
@@ -387,49 +387,52 @@
...
+		"detonator" = null,
+	)
+
 	if(src.holding)
-		data["holding"] = list()
-		data["holding"]["name"] = src.holding.name
-		data["holding"]["pressure"] = MIXTURE_PRESSURE(src.holding.air_contents)
-		data["holding"]["maxPressure"] = PORTABLE_ATMOS_MAX_RELEASE_PRESSURE
+		. += list(
+			"holding"= list(
```
Comment (mordent-goonstation) [inline code/modules/atmospherics/portable/canister.dm:403]:
```suggestion
			"holding" = list(
```

File: code/modules/atmospherics/portable/canister.dm [hunk 8]
```diff
@@ -387,53 +387,55 @@
...
+			)
+		)
 
 /obj/machinery/portable_atmospherics/canister/ui_static_data(mob/user)
-	var/list/static_data = list()
-
+	. = list(
+	"minRelease" = PORTABLE_ATMOS_MIN_RELEASE_PRESSURE,
+	"maxRelease" = PORTABLE_ATMOS_MAX_RELEASE_PRESSURE,
+	)
```
Comment (mordent-goonstation) [inline code/modules/atmospherics/portable/canister.dm:427]: Indent needs a fix. GitHub code suggestions are bad for this. 👀

File: tgui/packages/tgui/interfaces/GasTank.js [hunk 9]
```diff
@@ -1,8 +1,7 @@
-import { Fragment } from 'inferno';
 import { useBackend } from '../backend';
-import { Box, Section, AnimatedNumber } from '../components';
+import { Section, RoundGauge, LabeledList } from '../components';
```
Comment (mordent-goonstation) [inline tgui/packages/tgui/interfaces/GasTank.js:2]:
```suggestion
import { LabeledList, RoundGauge, Section } from '../components';
```
Pedantic? Yes. Alphabetizing was part of one of the tgui cleanup PR ports, though, so trying to keep us consistent there.

File: tgui/packages/tgui/interfaces/common/PortableAtmos.js [hunk 10]
```diff
@@ -1,6 +1,5 @@
-import { Fragment } from 'inferno';
-import { Box, Section, LabeledList, Button, AnimatedNumber } from '../../components';
-import { PressureBar } from './PressureBar';
+import { Box, Section, Button, RoundGauge, LabeledList } from '../../components';
```
Comment (mordent-goonstation) [inline tgui/packages/tgui/interfaces/common/PortableAtmos.js:1]:
```suggestion
import { Box, Button, LabeledList, RoundGauge, Section } from '../../components';
```

**Review by mordent-goonstation — CHANGES_REQUESTED** [review]

**Review by mordent-goonstation — APPROVED** [review]

**Pull request merged**